Refactor type checking of user-set variables

srstevenson opened the pull request:
Previously, the functions which checked the type of user-set configuration variables both checked the type of the variable, and printed an error message if the type was incorrect. In retrospect, this
is a mixing of responsibilities. Now, the functions just check the type of the variable passed to them and return a boolean. At the call site the value of this boolean is used to determine whether an error message is printed.

srstevenson committed:
Refactor type checking of user-set variables

Previously, the functions which checked the type of user-set
configuration variables both checked the type of the variable, and
printed an error message if the type was incorrect. In retrospect, this
is a mixing of responsibilities. Now, the functions just check the type
of the variable passed to them and return a boolean. At the call site
the value of this boolean is used to determine whether an error message
is printed.

srstevenson commented:
@boxofrox This implements what I mentioned in [my comment](https://github.com/srstevenson/vim-picker/pull/38/#issuecomment-451781981) on #38.

boxofrox commented on autoload/picker.vim:
```diff
@@ -271,31 +271,41 @@ function! s:PickFile(list_command, vim_command) abort
+    if type(a:variable) ==# type(0)
+        return v:true
+    else
+        return v:false
```
Maybe drop the `if` statements and just `return type(a:variable) ==# type(0)`?

Vim/Neovim doesn't have a type devoted to boolean.  `:help v:true` suggests the constant is there to support JSON encoding.  The commented boolean return value still applies even if the function returns 0/1 for false/true.

srstevenson committed:
Don't autoload type checking functions

Previously the functions to check the type of a variable were
autoloaded. However, because they were called in `plugin/picker.vim`,
`autoload/picker.vim` was sourced when Vim started, defeating the
purpose of autoloading. Now `plugin/picker.vim` doesn't call any
autoloaded functions and `autoload/picker.vim` is only sourced the first
time one of the fuzzy selection functions such as `picker#Split()` or
`picker#Tabedit()` is called.

srstevenson commented:
@boxofrox  I just noticed that autoloading these functions isn't a good idea, because they're called in `plugin/picker.vim` which is sourced on Vim start, and thus so is `autoload/picker.vim`, defeating the purpose of autoloading. I've moved them to `plugin/picker.vim` so `autoload/picker.vim` is only sourced the first     time one of the fuzzy selection functions such as `picker#Split()` or     `picker#Tabedit()` is called.
